### DISCUSSION TITLE
Integrating SHAP with OpenAI Whisper Model (medium.en) for Speech Transcription: Error with 'tokens' Argument

### DISCUSSION THREAD

**Discussion by Manishthakur2503** [openai/whisper#2266] [Q&A]
I've successfully used the OpenAI Whisper model (medium.en) for transcribing speech from audio files. Now, I'm trying to integrate SHAP (SHapley Additive exPlanations) for the first time to understand the model's predictions better. However, when attempting to generate SHAP explanations using the following code snippet:

```
def transcribe_audio(file_path=None, audio=None):
    global models_in_use, models

    while len(models_in_use) >= len(models):
        time.sleep(0.1)

    free_models = list(set(models.keys()) - set(models_in_use))
    transcribe_model = models[free_models[0]]
    models_in_use.append(free_models[0])

    if audio is not None:
        result = transcribe_model.transcribe(audio, condition_on_previous_text=False)
    else:
        result = transcribe_model.transcribe(file_path, condition_on_previous_text=False)

    transcription_text = result["text"]
    
    explanations = generate_shap_explanations(transcribe_model, transcription_text, audio=audio, file_path=file_path)

    models_in_use.remove(free_models[0])

    return transcription_text

def generate_shap_explanations(model, transcription_text, audio=None, file_path=None):
    if audio is not None:
        data = audio
    else:
        data = file_path

    # Create an explainer object
    explainer = shap.Explainer(model, tokenizer)

    # SHAP values calculation
    shap_values = explainer(transcription_text)
    
    return shap_values`

```

I encounter the following error:
TypeError: Whisper.forward() missing 1 required positional argument: 'tokens'
It seems that the Whisper model requires a 'tokens' argument, which I'm unsure how to provide when using SHAP for explanation. 

I'm also unsure whether SHAP can be used effectively with the Whisper model for speech transcription. Could you please advise on how to correctly integrate SHAP with the Whisper model for generating explanations of its predictions, and confirm if SHAP is suitable for this use case? Thank you!

**Comment by phineas-pta** [comment]
it doesn't support ASR yet
see shap/shap#2552

**Comment by Manishthakur2503** [comment]
@phineas-pta Thanks for the information
